OrbitCore/TracepointEventBuffer.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointEventBuffer.h"
+
+using orbit_client_protos::TracepointEventInfo;
+
+std::vector<TracepointEventInfo> TracepointEventBuffer::GetTracepointEvents(
+    uint64_t time_begin, uint64_t time_end, int32_t thread_id /*= kAllThreadsFakeTid*/) {
+  ScopeLock lock(mutex_);
+  std::vector<TracepointEventInfo> tracepoint_events;
+  for (auto& pair : tracepoint_events_) {
+    const int32_t tracepoint_thread_id = pair.first;
+    std::map<uint64_t, TracepointEventInfo>& tracepoints = pair.second;
+
+    CHECK(thread_id == SamplingProfiler::kAllThreadsFakeTid || tracepoint_thread_id == thread_id);
```

Comment:
I'm confused about this CHECK. Isn't it normal that `tracepoint_thread_id != thread_id` and you would just skip to the next iteration? Have a look at this loop in `EventBuffer::GetCallstackEventsInTimeRangeForThreadId`. You might want a condition here instead (`if (thread_id == SamplingProfiler::kAllThreadsFakeTid || callstack_thread_id == thread_id)`).